plugins/content-manager/src/main/java/com/wazuh/contentmanager/index/CTIConsumers.java

```diff
@@ -154,27 +158,41 @@ private static long asLong(Object o) {
     }
 
     /**
-     * Checks whether the {@link ContextIndex#INDEX_NAME} index exists.
+     * Checks whether the {@link CTIConsumers#INDEX_NAME} index exists.
      *
      * @see ClusterInfo#indexExists(Client, String)
      * @return true if the index exists, false otherwise.
      */
     public boolean exists() {
-        return ClusterInfo.indexExists(this.client, ContextIndex.INDEX_NAME);
+        return ClusterInfo.indexExists(this.client, CTIConsumers.INDEX_NAME);
     }
 
-    /** Creates the {@link ContextIndex#INDEX_NAME} index, if it does not exist. */
+    /** Creates the {@link CTIConsumers#INDEX_NAME} index, if it does not exist. */
     public void createIndex() {
         if (!this.exists()) {
-            boolean result =
-                    this.index(
-                            new ConsumerInfo(
-                                    this.pluginSettings.getConsumerId(),
-                                    this.pluginSettings.getContextId(),
-                                    0,
-                                    0,
-                                    null));
-            log.info("Index initialized: {}", result);
+            try {
+                CreateIndexRequest request = new CreateIndexRequest(CTIConsumers.INDEX_NAME);
+                Settings settings = Settings.builder().put("index.number_of_replicas", 0).build();
+                request.settings(settings);
+                String mapping = "{\n"
+                    + "  \"properties\": {\n"
+                    + "    \"rules_consumer\": {\n"
+                    + "      \"properties\": {\n"
+                    + "        \"current_offset\": { \"type\": \"long\" },\n"
+                    + "        \"latest_offset\": { \"type\": \"long\" },\n"
+                    + "        \"snapshot_link\": { \"type\": \"keyword\" }\n"
+                    + "      }\n"
+                    + "    }\n"
+                    + "  }\n"
+                    + "}";
```

Comment:
`rules_consumer` is the name of the consumer, which is dynamic.
Also, try to extract the mapping to a file instead.